Makefile

```diff
@@ -0,0 +1,127 @@
+DC_CI = "bin/dc.sh"
+DC = $(shell which docker-compose)
+
+default: help
+	@echo ""
+	@echo "You need to specify a subcommand."
+	@exit 1
+
+help:
+	@echo "build         - build docker images for dev"
+	@echo "run           - docker-compose up the entire system for dev"
+	@echo ""
+	@echo "pull          - pull the latest production images from Docker Hub"
+	@echo "init          - initialize the database and install Node and Bower packages"
+	@echo "djshell       - start a Django Python shell (ipython)"
+	@echo "dbshell       - start a MySQL shell"
+	@echo "shell         - start a bash shell"
+	@echo "clean         - remove all build, test, coverage and Python artifacts"
+	@echo "rebuild       - force a rebuild of all of the docker images"
+	@echo "lint          - check style with flake8, jshint, and stylelint"
+	@echo "test          - run tests against local files"
+	@echo "test-image    - run tests against files in docker image"
+	@echo "docs          - generate Sphinx HTML documentation"
+	@echo "build-ci      - build docker images for use in our CI pipeline"
+	@echo "test-ci       - run tests against files in docker image built by CI"
+
+.env:
+	@if [ ! -f .env ]; then \
+		echo "Copying .env-dist to .env..."; \
+		cp .env-dist .env; \
+	fi
+
+.docker-build:
+	${MAKE} build
+
+.docker-build-pull:
+	${MAKE} pull
+
+.docker-build-full:
+	${MAKE} build-full
+
+build: .docker-build-pull
+	${DC} build base-dev
+	touch .docker-build
+
+build-full: .docker-build-pull
+	${DC} build full
+#   tag other images
+	${DC} build base base-dev staticfiles locales full-no-locales
+	touch .docker-build-full
+
+pull: .env
+	-GIT_COMMIT= ${DC} pull base base-dev staticfiles locales full-no-locales full mariadb elasticsearch redis
+	touch .docker-build-pull
+
+rebuild: clean build
+
+run: .docker-build-pull
+	${DC} up web
```

Comment:
Personally I prefer to use

`docker-compose run web --service-ports bash`

and then in the shell run the server.

I believe this way you can debug better and also restart easier when things fail.

I'm not a huge user of DC though and i'd love comments on this.

Reply:
huh... that's a good idea. I've never done it that way. I don't see why that wouldn't work, but it is a bit more to do. Perhaps it would be best to document this and add `--service-ports` to the `make shell` command if people want to do this?

Reply:
You can only add `--service-ports` to one active shell command, otherwise DC will complain that it can't bind to the ports. So not sure if changing `make shell` is the right thing to do. 

That said feel free to go whichever way you prefer, I just dropped an idea :)

Reply:
@glogiotatidis thanks for the suggestion! I added a note in the README about this technique and added `--service-ports` to the `make shell` command. I think this way both methods are well supported.

Reply:
hmm... I assume it's complaining in the same way that anything complains when ports are in use? I guess it's possible you'd want a shell while you already have other things running. I could instead make another command for this specificially; `make runshell` maybe. That way you could decide whether or not you wanted that. I could also be making the makefile do too much and we could just document how to use docker-compose directly in this case. Not sure.

Reply:
I updated things to add a `make runshell` command that binds ports, and removed port binding from `make shell`. I think that's enough make automation. Anything else can be done with one of those two commands. Thoughts?

Reply:
Yap, `runshell` is probably the best thing. thanks for taking the time